### PR TITLE
NW6-Bakhat begum-JS3-Module-Project-main-sprint-3

### PR DESCRIPTION
**Pull request by BakhatBegum**
## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

**Comment by rafsanjani** [comment]
Can you resolve these conflicts? If you need assistance resolving them cleanly, please do mention.

File: script.js [hunk 1]
```diff
@@ -1,48 +1,80 @@
...
+  }
+  
 
 function setup() {
-  const allEpisodes = state.getAllEpisodes;
-  makePageForEpisodes(allEpisodes);
-  const numb = document.getElementById("num-epis");
-  numb.textContent = ` ${allEpisodes.length} `;
-  render();
+    fetchFilms().then(function (films) {
```
Comment (rafsanjani) [inline script.js]: You could use arrow functions here, just like you did in `fetchShows`

File: script.js [hunk 2]
```diff
@@ -1,48 +1,80 @@
...
+      render();
+    });
+  });
 }
 
 function makePageForEpisodes(episodeList) {
   const root = document.getElementById("root");
   root.innerHTML = "";
-  for (const episode of episodeList) {
+ // for (const episode of episodeList) {
```
Comment (rafsanjani) [inline script.js]: Can you delete the commented code completely? When working with git, we can always rollback changes or time travel into the past if something goes wrong.

File: script.js [hunk 3]
```diff
@@ -62,4 +94,94 @@ function render() {
...
 }
 
+const selectEpisodes = document.getElementById("select-episode")
+function createDropdownOptions(episodeList){
+  selectEpisodes.innerHTML = "";
+
+  const defaultOption = document.createElement("option");
+  defaultOption.value = "";
+  defaultOption.innerHTML = "Select Episodes";
+  selectEpisodes.appendChild(defaultOption);
```
Comment (rafsanjani) [inline script.js]: If you could put an empty line between lines `104` and `105`, it would make the code look more readable

**Comment by BakhatBegum** [comment]
const getEpisodes = document.getElementById("selectEpisodes");
const getShow = document.getElementById("selectShow");
const root = document.getElementById("root");
const allShows = [];
let specialEp = [];

function fetchShows() {
  return fetch("https://api.tvmaze.com/shows")
    .then((res) => {
      if (!res.ok) {
        throw new Error("Failed to fetch shows");
      }
      return res.json();
    })
    .then((data) => {
      for (const show of data) {
        const episodeDiv = document.createElement("div");
        episodeDiv.setAttribute("id", "container");
        displayEpisodes(show, episodeDiv);
        root.appendChild(episodeDiv);
        allShows.push(show);
      }
      getAllShows();
      displaywithTitle();
    })
    .catch((err) => {
      console.error(err);
      return [];
    });
}
fetchShows();
// console.log(allShows);
function getAllShows() {
  getEpisodes.innerHTML = "";
  for (nameShows of allShows) {
    const option = document.createElement("option");
    option.textContent = nameShows.name;
    // console.log("name =====>", nameShows.name);
    getEpisodes.appendChild(option);
  }
}

getEpisodes.addEventListener("change", function () {
  const showValue = this.value;
  console.log("Selected Show:", showValue);

  // Clear previous episodes and specific episodes
  root.innerHTML = "";
  getShow.innerHTML = "";
  specialEp = [];

  // Fetch episodes for the selected show
  const selectedShow = allShows.find((show) => show.name.includes(showValue
));
  if (selectedShow) {
    fetchEpisodes(selectedShow.id);
  }
});

getShow.addEventListener("change", function () {
  const selectedSpecificEpisode = this.value;
  console.log("Selected Specific Episode:", selectedSpecificEpisode);

  // Clear previous episodes
  root.innerHTML = "";

  // Display episodes for the selected specific episode
  for (const episodes of specialEp) {
    for (const episode of episodes) {
      if (episode.name.includes(selectedSpecificEpisode)) {
        const episodeDiv = document.createElement("div");
        episodeDiv.setAttribute("id", "container");
        displayEpisodes(episode, episodeDiv);
        root.appendChild(episodeDiv);
      }
    }
  }
});

function displayEpisodes(episode, episodeDiv) {
  episodeDiv.innerHTML = `
      <h2 class="forEpisode">${episode.name} </h2>
      ${episode.name}
      <p>${episode.summary}</p>
    `;
}

function fetchEpisodes(showId) {
  return fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
    .then((response) => {
      if (!response.ok) {
        throw new Error("Failed to fetch episodes");
      }
      return response.json();
    })
    .then((data) => {
      root.innerHTML = "";
      specialEp.push(data);
      console.log(data);
      for (show of data) {
        const episodeDiv = document.createElement("div");
        episodeDiv.setAttribute("id", "container");
        displayEpisodes(show, episodeDiv);
        root.appendChild(episodeDiv);
        const option = document.createElement("option");
        option.textContent = show.name;
        getShow.appendChild(option);
      }
    })
    .catch((error) => {
      getShow.innerHTML = "";
      console.error("Error fetching episodes:", error);
    });
}

function displaywithTitle() {
  const h2Elements = root.querySelectorAll(".forEpisode");
  for (h2 of h2Elements) {
    // h2Elements.forEach((h2) => {
    const can = h2.textContent;
    can.addEventListener("click", function () {
      const offf = this.value;
      console.log(offf);
    });
  }
  // });
}

**Pull request merged**